tests/models/test_taskinstance.py

```diff
@@ -1770,6 +1770,59 @@ def test_get_num_running_task_instances(self, create_task_instance):
         assert 1 == ti2.get_num_running_task_instances(session=session)
         assert 1 == ti3.get_num_running_task_instances(session=session)
 
+    def test_get_num_running_task_instances_per_dagrun(self, create_task_instance, dag_maker):
+        session = settings.Session()
+
+        @task
+        def dummy_mapped_task(x: int):
+            return x
+
+        with dag_maker(dag_id="test_dag") as dag:
+            dummy_mapped_task.expand(x=[1, 2])
+            dagrun1 = dag.create_dagrun(execution_date=timezone.utcnow(),
+                                       state=DagRunState.RUNNING,
+                                       run_id="run_id_1",
+                                       session=session)
+
+            ti1 = dagrun1.task_instances[0]
```

Comment:
`task_instances` is not ordered so this test would be flaky as is. A better pattern would be something like

```python
tis = {(ti.task_id, ti.map_index): ti for ti in dr.task_instances}
```

And use the dict to look up a specific task instance.